pkg/agent/downloader.go

```diff
@@ -0,0 +1,85 @@
+package agent
+
+import (
+	"fmt"
+	pro "github.com/kubeflow/kfserving/pkg/agent/protocols"
```

Comment:
nit: I'd probably just leave this without a name and let it default to the package name "protocols"